pkg/features/oidc.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2019 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package features
+
+import (
+	kubeadmv1beta1 "github.com/kubermatic/kubeone/pkg/apis/kubeadm/v1beta1"
+	"github.com/kubermatic/kubeone/pkg/config"
+)
+
+func activateKubeadmOIDC(feature config.OpenIDConnect, cfg *kubeadmv1beta1.ClusterConfiguration) {
+	if !feature.Enable {
+		return
+	}
+
+	if cfg.APIServer.ExtraArgs == nil {
+		cfg.APIServer.ExtraArgs = map[string]string{}
+	}
+
+	cfg.APIServer.ExtraArgs["oidc-issuer-url"] = feature.Config.IssuerURL
+	cfg.APIServer.ExtraArgs["oidc-client-id"] = feature.Config.ClientID
+	optinalMapSet(cfg.APIServer.ExtraArgs, "oidc-username-claim", feature.Config.UsernameClaim)
+	optinalMapSet(cfg.APIServer.ExtraArgs, "oidc-username-prefix", feature.Config.UsernamePrefix)
+	optinalMapSet(cfg.APIServer.ExtraArgs, "oidc-groups-claim", feature.Config.GroupsClaim)
+	optinalMapSet(cfg.APIServer.ExtraArgs, "oidc-groups-prefix", feature.Config.GroupsPrefix)
+	optinalMapSet(cfg.APIServer.ExtraArgs, "oidc-required-claim", feature.Config.RequiredClaim)
+	optinalMapSet(cfg.APIServer.ExtraArgs, "oidc-signing-algs", feature.Config.SigningAlgs)
+	optinalMapSet(cfg.APIServer.ExtraArgs, "oidc-ca-file", feature.Config.CAFile)
+}
+
+func optinalMapSet(m map[string]string, key string, val string) {
```

Comment:
What does `optinalMapSet` mean? Did you mean `optionalMapSet`?

Reply:
typo 🤷‍♂️, fixed: ✔️